lib/ddtrace/notifications.rb

```diff
@@ -0,0 +1,34 @@
+module Datadog
+  class Notifications
+    def self.subscribe(pattern, tracer = Datadog.tracer, &block)
+      subscriber = Subscriber.new(tracer, &block)
+
+      ActiveSupport::Notifications.subscribe(pattern, subscriber)
+    end
+
+    class Subscriber
+      def initialize(tracer, &block)
+        @tracer = tracer
+        @block = block
+
+        # A stack of open spans. We want to get to the most recently opened span
+        # at any point in time.
+        @spans = []
+      end
+
+      def start(_name, _id, _payload)
+        @spans << @tracer.trace("<tmp>")
+      end
+
+      def finish(name, id, payload)
+        # We close spans in reverse order.
+        span = @spans.pop or raise "no spans left in stack!"
```

Comment:
It's thoughtful to use a stack here, but is this thread safe? Haven't been able to confirm that much. We do have an `active_span` function for retrieving spans in a safe way, which might be a good substitute for the stack here.

The equivalent would be:

```ruby
span = Datadog.tracer.active_span
```

Reply:
I believe it's thread safe, as AS::N subscribers are thread local, but I'll switch to this way.

Reply:
Hmm, that doesn't seem to work...

Reply:
What kind of problem did you run into?